nui/src/main/java/org/terasology/nui/widgets/types/builtin/util/GrowableListWidgetBuilder.java

```diff
@@ -0,0 +1,215 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+package org.terasology.nui.widgets.types.builtin.util;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.nui.UIWidget;
+import org.terasology.nui.WidgetUtil;
+import org.terasology.nui.databinding.Binding;
+import org.terasology.nui.databinding.DefaultBinding;
+import org.terasology.nui.databinding.NotifyingBinding;
+import org.terasology.nui.layouts.ColumnLayout;
+import org.terasology.nui.layouts.RowLayout;
+import org.terasology.nui.layouts.RowLayoutHint;
+import org.terasology.nui.widgets.UIButton;
+import org.terasology.nui.widgets.UILabel;
+import org.terasology.nui.widgets.UISpace;
+import org.terasology.nui.widgets.types.TypeWidgetBuilder;
+import org.terasology.nui.widgets.types.TypeWidgetLibrary;
+import org.terasology.reflection.ReflectionUtil;
+import org.terasology.reflection.TypeInfo;
+import org.terasology.reflection.reflect.ObjectConstructor;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.terasology.nui.widgets.types.TypeWidgetFactory.LABEL_WIDGET_ID;
+
+public abstract class GrowableListWidgetBuilder<C, E> implements TypeWidgetBuilder<C> {
```

Comment:
Is this builder also in place for the `Set<String>` property on the example screen? 

I noticed that the "Add" button stays in place when adding elements (elements are added below), but after collapsing and expanding the list view the "Add" button moves to the end of the list. When adding more elements the button will be in the middle, because new elements are added below.

![image](https://user-images.githubusercontent.com/1448874/108771841-78428b00-755c-11eb-839c-4b88906202a6.png)

Reply:
Fixed. it just standart problem with addLast in list :)